h3ronpy/python/h3ronpy/arrow/__init__.py

```diff
@@ -1,35 +1,50 @@
-from typing import Union
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Optional, Sequence, Union, cast
+
+from arro3.core import Array, ChunkedArray, DataType, RecordBatch
+from arro3.core.types import (
+    ArrowArrayExportable,
+    ArrowSchemaExportable,
+    ArrowStreamExportable,
+)
 
-import pyarrow as pa
 from h3ronpy.h3ronpyrs import op
 
-try:
+if TYPE_CHECKING:
     import polars as pl
 
-    _HAS_POLARS = True
-except ImportError:
-    _HAS_POLARS = False
 
+def _to_arrow_array(
+    arr: Union[ArrowArrayExportable, ArrowStreamExportable, pl.Series, Sequence[Any]],
+    dtype: Optional[ArrowSchemaExportable] = None,
+) -> Array:
+    if hasattr(arr, "__arrow_c_array__"):
+        array = Array.from_arrow(cast(ArrowArrayExportable, arr))
+    elif hasattr(arr, "__arrow_c_stream__"):
```

Comment:
What was reason you preferred doing the conversion from the stream interface to the array interface in python instead of using the `pyo3-arrow` `AnyArray`-enum and converting in rust?

Reply:
`AnyArray` won't look for a `to_arrow` method on an object. If we want to support input from older versions of pandas or polars that doesn't implement the PyCapsule Interface, then we'll need to have some sort of Python wrapper here.

Reply:
Alright - good choice using the python conversion in that case.